name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSet.java

```diff
@@ -68,6 +81,48 @@ public DataSeries lookup(String uuid)
         return availableSeries.stream().filter(d -> d.getUUID().equals(uuid)).findAny().orElse(null);
     }
 
+    private void buildStatementOfAssetsDerivedDataSeries(ColorWheel wheel)
+    {
+        for (var baseDataSeries : availableSeries)
+        {
+            // skip client - the client already defines all derived data series
+            // (but only for the total client)
+            if (baseDataSeries.getType() == DataSeries.Type.CLIENT)
+                continue;
+
+            for (var type : DataSeries.ClientDataSeries.values())
+            {
+                // skip totals - that is already covered by the base data series
+                // itself
+                if (type == ClientDataSeries.TOTALS)
+                    continue;
+                // skip delta percentage - that is applicable only for
+                // performance data series
+                if (type == ClientDataSeries.DELTA_PERCENTAGE)
+                    continue; // NOSONAR
+
+                // skip interest and interest charge for portfolios and
+                // securities - not supported
+                if ((baseDataSeries.getType() == DataSeries.Type.SECURITY
+                                || baseDataSeries.getType() == DataSeries.Type.PORTFOLIO)
+                                && EnumSet.of(ClientDataSeries.INTEREST, ClientDataSeries.INTEREST_ACCUMULATED,
+                                                ClientDataSeries.INTEREST_CHARGE,
+                                                ClientDataSeries.INTEREST_CHARGE_ACCUMULATED).contains(type))
+                    continue;
+
+                var label = type.getLabel() + ": " + baseDataSeries.getLabel(); //$NON-NLS-1$
+
+                var dataSeries = new DataSeries(DataSeries.Type.DERIVED_DATA_SERIES, baseDataSeries.getGroup(),
+                                new DerivedDataSeries(baseDataSeries, type), label, wheel.next());
+
+                dataSeries.setLineChart(baseDataSeries.isLineChart());
+                dataSeries.setShowArea(baseDataSeries.isShowArea());
```

Comment:
here : it will always be a Line without Area. we can check ClientDataSeries types before and assign a correct boolean for Line and Area ?